src/Symfony/Component/Form/Tests/Extension/Core/Type/CheckboxTypeTest.php

```diff
@@ -170,4 +170,38 @@ public function provideCustomModelTransformerData()
             array('unchecked', false),
         );
     }
+
+    public function testSubmitNullWithEmptyDataTrueWhenNested()
+    {
+        $form = $this->factory->create('form')
```

Comment:
Do the tests pass (before the change) if this is not nested? If not, I'd simplify this test.